### PR TITLE
Infinite loading on video preview before sending video

### PR DESCRIPTION
**Pull request by ikevin127**
### Details

Due to a WebKit browser engine limitation which does not allow us to skip the first milisecond of the video blob programatically via `#t=` (time fragment), we removed that functionality specifically for if `Browser.isMobileWebKit()` is true which means only for iOS: mWeb.

### Fixed Issues

$ https://github.com/Expensify/App/issues/39078
PROPOSAL: https://github.com/Expensify/App/issues/39078#issuecomment-2032602290

### Tests

1. Go to any chat.
2. Tap on "+" button.
3. Select "Add attachment" and any other option.
4. Select a video.
5. Verify there's no more infinite loading of video on iOS: mWeb and the video preview is transparent.
6. The behaviour on all other platforms should be: no infinite loading and additionally the video preview will show as expected.

- [x] Verify that no errors appear in the JS console

### Offline tests

**TLDR:** same as Tests.

1. Go to any chat.
2. Tap on "+" button.
3. Select "Add attachment" and any other option.
4. Select a video.
5. While offline: there will always be an infinite loading indicator, on all platforms.

### QA Steps

**TLDR:** same as Tests.

1. Go to any chat.
2. Tap on "+" button.
3. Select "Add attachment" and any other option.
4. Select a video.
5. Verify there's no more infinite loading of video on iOS: mWeb and the video preview is transparent.
6. The behaviour on all other platforms should be: no infinite loading and additionally the video preview will show as expected.

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/56457735/9620bc8d-3193-45a9-b6ed-d5269525b131

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/56457735/2b04e8e2-88d5-419f-ba80-d17ce5ce7adb

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/56457735/48c2ece8-d334-4a4f-939e-d6d957cf52c2

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/56457735/857c714b-15b1-42a5-abf6-d7124e33891c

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/56457735/fdbff217-f7c6-48fd-9cb4-6e7f756c9c86

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/56457735/7447f6a6-4a98-4868-93df-06e291785f72

</details>

### CODE REVIEW

**Comment by ikevin127** [comment]
> @ikevin127 I just tested your PR on both Safari & Chrome on a physical iPhone – it works fine. Can you reliably reproduce the issue you were talking about https://github.com/Expensify/App/issues/39078#issuecomment-2041023465?

@paultsimura It still happens on my side consistently. But if it does not happen on your side then there must be something wrong with my physical iPhone - therefore we can move on with this solution 👍 I re-checked the **PR Author Checklist**.

In case we get a future regression based on https://github.com/Expensify/App/issues/39078#issuecomment-2041023465, then we can apply the 2nd proposed solution and we'll know it will work for sure because it's not using play / pause.

File: src/libs/Browser/index.website.ts [hunk 1]
```diff
@@ -101,4 +109,4 @@ const openRouteInDesktopApp: OpenRouteInDesktopApp = (shortLivedAuthToken = '',
     }
 };
 
-export {getBrowser, isMobile, isMobileSafari, isSafari, isMobileChrome, openRouteInDesktopApp};
+export {getBrowser, isMobile, isMobileSafari, isMobileWebKit, isSafari, isMobileChrome, openRouteInDesktopApp};
```
[the remark this answers is not in the document]
Comment (ikevin127) [reply src/libs/Browser/index.website.ts:112]: Nope, we need to cover all iOS: mWeb WebKit-based browser engines, this includes iOS: Chrome, Firefox and ay other browsers which are forced by Apple to use the WebKit engine.

Therefore we need to cover for all of them, otherwise the infinite loading would appear on Chrome / Firefox / other browsers one can install on their iPhone.

**Review by ishpaul777 — APPROVED** [review]

**Review by amyevans — APPROVED** [review]

**Comment by kbecciv** [comment]
@ikevin127 @amyevans I'm a bit confused about the Offline steps. It says there should be 'no infinite loading,' but when following the steps offline, there's still an infinite loading screen. Is this expected behavior?
image

https://github.com/Expensify/App/assets/93399543/23403080-5fc1-468c-9ccd-525a54fea3df

**Comment by ikevin127** [comment]
@kbecciv My bad, I just copy & pasted the same tests from online without checking. I corrected the offline test just now!

**Note:** It is expected behaviour that we see the infinite loading while **offline** since this was the behaviour before the PR was merged (can be verified on production).
